sklearn/model_selection/_validation.py

```diff
@@ -992,10 +1136,60 @@ def validation_curve(estimator, X, y, param_name, param_range, groups=None,
         parameters={param_name: v}, fit_params=None, return_train_score=True)
         # NOTE do not change order of iteration to allow one time cv splitters
         for train, test in cv.split(X, y, groups) for v in param_range)
+    out = list(zip(*out))
 
-    out = np.asarray(out)
     n_params = len(param_range)
-    n_cv_folds = out.shape[0] // n_params
-    out = out.reshape(n_cv_folds, n_params, 2).transpose((2, 1, 0))
+    n_splits = cv.get_n_splits(X, y)
+
+    out[0] = (np.array(out[0]).reshape(n_splits, n_params)).T
+    out[1] = (np.array(out[1]).reshape(n_splits, n_params)).T
 
     return out[0], out[1]
+
+
+def _aggregate_score_dicts(scores, shape=None, transpose=False):
+    """Aggregate the list of dict to dict of np ndarray
+
+    The aggregated output of _fit_and_score will be a list of dict
+    of form [{'prec': 0.1, 'acc':1.0}, {'prec': 0.1, 'acc':1.0}, ...]
+    Convert it to a dict of array {'prec': np.array([0.1 ...]), ...}
+
+    If shape is set, the individual numpy arrays are reshaped accordingly
+
+    If transpose is set, the individual arrays are transposed after reshaping
+    This is useful if the the scores are not ordered first by cv splits.
+
+    Example
+    -------
+
+    >>> scores = [{'a': 1, 'b':1}, {'a': 2, 'b':2}, {'a': 3, 'b':3},
+    ...           {'a': 4, 'b':4}, {'a': 5, 'b':5}, {'a': 6, 'b':6},
+    ...           {'a': 7, 'b':7}, {'a': 8, 'b':8}, {'a': 9, 'b':9},
+    ...           {'a': 10, 'b': 10}]                          # doctest: +SKIP
+    >>> _aggregate_score_dicts(scores,                         # doctest: +SKIP
+    ...                        shape=(2, 5), transpose=False)
+    {'a': array([[1, 2, 3, 4, 5],
+                 [6, 7, 8, 9, 10]]),
+     'b': array([[1, 2, 3, 4, 5],
+                 [6, 7, 8, 9, 10]])}
+    >>> _aggregate_score_dicts(scores,                         # doctest: +SKIP
+    ...                        shape=(2, 5), transpose=True)
+    {'a': array([[1, 6], [2, 7], [3, 8], [4, 9], [5, 10]]),
+     'b': array([[1, 6], [2, 7], [3, 8], [4, 9], [5, 10]])}
+    """
+    if shape is None:
+        shape = (len(scores), )
+    else:
+        if np.product(shape) != len(scores):
+            raise ValueError("Shape must conform to the size of the scores")
+
+    out = {}
+
+    for key in scores[0]:
+        key_scores = np.asarray([score[key] for score in scores])
+        key_scores = key_scores.reshape(shape)
+        if transpose:
+            key_scores = key_scores.T
```

Comment:
True. But I can't think of a better way to do this. Maybe it should not be shape and rather renamed to `order`?